### PR TITLE
xds/client: Process security configuration from Cluster resource

### PR DESCRIPTION
**Pull request by easwars**

### CODE REVIEW

File: xds/internal/version/version.go
Comment (menghanl) [inline xds/internal/version/version.go:48]: Why is this line a `v2`, not `v3`?

File: xds/internal/client/client_xds.go
Comment (menghanl) [inline xds/internal/client/client_xds.go:421]: How would the CDS balancer know if this field exists in the CDS response? Does it check for empty strings? Or make this field a pointer?

File: xds/internal/version/version.go
Comment (easwars) [reply xds/internal/version/version.go]: Typo from earlier I would guess. Fixed.

File: xds/internal/client/client_xds.go
Comment (easwars) [reply xds/internal/client/client_xds.go]: Done. Changed it to a pointer. I initially had it as a pointer, then saw that none of the `Update` types had a pointer, so switched it to value. But pointer definitely is better here. Thanks.

**Comment by easwars** [comment]
Done. PTAL.

**Review by menghanl — APPROVED** [review]

File: go.mod
Comment (menghanl) [inline go.mod:9]: The go.mod change and vet.sh will conflict with #3932 
The other PR is mostly generated code, and has a cleaner change in vet.sh, maybe wait for that to merge first?

Comment (easwars) [reply go.mod:9]: Yes, will wait for it.

Comment (menghanl) [reply go.mod:9]: And even after that, merging go.mod/go.sum won't be pleasant I guess. I would try first revert those changes, then merge and `go mod tidy`.

Comment (easwars) [reply go.mod:9]: I had to add a few more lines to `vet.sh` from the ones added in #3932 because my changes here do not have the corresponding pb.go changes. So, there were more complaints about usage of deprecated fields/types. We probably should do a scrub of the list once #3932 gets in.

**Pull request merged**
